test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -130,6 +148,156 @@ public void checkDirtyReads() throws Exception {
+            txn.delete(table, key1);
+            txn.commit();
+        }
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1)).isNull();
+        expectedVersion = 0L;
+
+        try (ManagedTxnContext txn = corfuStore.tx(nsxManager)) {
+            txn.putRecord(table, key1, SampleSchema.EventInfo.newBuilder().setName("def").build(), user_2);
+            txn.commit();
+        }
+
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1).getMetadata())
+                .isEqualTo(ManagedResources.newBuilder(user_2)
+                        .setCreateTimestamp(0L)
+                        .setNestedType(SampleSchema.NestedTypeA.newBuilder().build())
+                        .setVersion(expectedVersion).build());
+
+        // Validate that if Metadata schema is specified, a null metadata is not acceptable
+        assertThatThrownBy(() ->
+                table.update(key1, SampleSchema.EventInfo.getDefaultInstance(), null))
+                .isExactlyInstanceOf(RuntimeException.class);
+
+        // Validate that we throw a StaleObject exception if there is an explicit version mismatch
+        ManagedResources wrongRevisionMetadata = ManagedResources.newBuilder(user_2)
+                .setVersion(2).build();
+
+        assertThatThrownBy(() ->
+                table.update(key1, SampleSchema.EventInfo.getDefaultInstance(), wrongRevisionMetadata))
+                .isExactlyInstanceOf(RuntimeException.class);
+
+        // Verify the table is readable using entryStream()
+        final int batchSize = 50;
+        Stream<CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.EventInfo, ManagedResources>> entryStream = table.entryStream();
+        final Iterable<List<CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.EventInfo, ManagedResources>>> partitions =
+                Iterables.partition(entryStream::iterator, batchSize);
+        for (List<CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.EventInfo, ManagedResources>> partition : partitions) {
+            for (CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.EventInfo, ManagedResources> entry : partition) {
+                assertThat(entry.getKey()).isExactlyInstanceOf(SampleSchema.Uuid.class);
+                assertThat(entry.getPayload()).isExactlyInstanceOf(SampleSchema.EventInfo.class);
+                assertThat(entry.getMetadata()).isExactlyInstanceOf(ManagedResources.class);
+            }
+        }
+    }
+
+    /**
+     * Demonstrates that opening same table from multiple threads will retry internal transactions
+     *
+     * @throws Exception
+     */
+    @Test
+    public void checkOpenRetriesTXN() throws Exception {
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+        CorfuStoreShim corfuStore = new CorfuStoreShim(corfuRuntime);
+        final String nsxManager = "nsx-manager"; // namespace for the table
+        final String tableName = "EventInfo"; // table name
+        final int numThreads = 10;
```

Comment:
not related to your change but this seems to be a source of flakiness in tests, because in GitHub actions the number of cores may be far less, executing 10 threads can sometimes timeout, perhaps we can reduce this to 5?